### ISSUE TITLE
Update example to use a validation set

### ISSUE THREAD

**Issue by ethanwharris** [ecs-vlc/torchbearer#239]
Currently uses Cifar10 test set for validation, leads to overfitting. Update to put in a train/val split

**Comment by ethanwharris** [comment]
@MattPainter01 Would be nice to make this easier, perhaps have a val_split option in the main fit calls? (with either a seed or the split stored in state so it persists when saved???)
